src/pages/[platform]/build-a-backend/data/custom-business-logic/search-and-aggregate-queries/index.mdx

```diff
@@ -0,0 +1,431 @@
+
+```ts title="amplify/backend.ts"
+
+import { defineBackend } from "@aws-amplify/backend";
+import { auth } from "./auth/resource";
+import { data } from "./data/resource";
+import * as dynamodb from "aws-cdk-lib/aws-dynamodb";
+
+const backend = defineBackend({
+  auth,
+  data,
+});
+
+const todoTable =
+  backend.data.resources.cfnResources.amplifyDynamoDbTables["Todo"];
+
+todoTable.pointInTimeRecoveryEnabled = true;
+
+todoTable.streamSpecification = {
+  streamViewType: dynamodb.StreamViewType.NEW_IMAGE,
+};
+
+```
+
+## Step 2: Setting Up the OpenSearch Instance
+
+Create an OpenSearch instance with encryption.
+
+``` ts title="amplify/backend.ts"
+const openSearchDomain = new opensearch.Domain(
+  openSearchStack,
+  "OpenSearchDomain",
+  {
+    version: opensearch.EngineVersion.OPENSEARCH_2_11,
+    nodeToNodeEncryption: true,
+    encryptionAtRest: {
+      enabled: true,
+    },
+  }
+);
+
+```
+
+## Step 3: Setting Up Zero ETL from DynamoDB to OpenSearch
+
+### Step 3a: S3 Bucket and IAM Role
+Establish an S3 bucket to back up raw events consumed by the OpenSearch pipeline. Additionally, configure an IAM role for the pipeline and assign the roles as indicated below. For further information on the required IAM roles, please refer to the [Setting up roles and users](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/pipeline-security-overview.html#pipeline-security-create) documentation.
+
+``` ts title="amplify/backend.ts"
+import { defineBackend } from "@aws-amplify/backend";
+import { auth } from "./auth/resource";
+import { data } from "./data/resource";
+import * as dynamodb from "aws-cdk-lib/aws-dynamodb";
+import * as opensearch from "aws-cdk-lib/aws-opensearchservice";
+import * as osis from "aws-cdk-lib/aws-osis";
+import * as s3 from "aws-cdk-lib/aws-s3";
+import * as iam from "aws-cdk-lib/aws-iam";
+import * as logs from "aws-cdk-lib/aws-logs";
+import { RemovalPolicy } from "aws-cdk-lib";
+import { PolicyStatement } from "aws-cdk-lib/aws-iam";
+
+const s3BackupBucket = new s3.Bucket(
+  openSearchStack,
+  "OpenSearchBackupBucketAmplifyGen2",
+  {
+    blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
+    bucketName: "opensearch-backup-bucket-amplify-gen-2-test",
+    enforceSSL: true,
+    versioned: true,
+    autoDeleteObjects: true,
+    removalPolicy: RemovalPolicy.DESTROY,
+  }
+);
+
+const openSearchIntegrationPipelineRole = new iam.Role(
+  openSearchStack,
+  "OpenSearchIntegrationPipelineRole",
+  {
+    assumedBy: new iam.ServicePrincipal("osis-pipelines.amazonaws.com"),
+    inlinePolicies: {
+      openSearchPipelinePolicy: new iam.PolicyDocument({
+        statements: [
+          new iam.PolicyStatement({
+            actions: ["es:DescribeDomain"],
+            resources: ["<openSearchDomain-arn>"],
+            effect: iam.Effect.ALLOW,
+          }),
+          new iam.PolicyStatement({
+            actions: ["es:ESHttp*"],
+            resources: ["*"],
```

Comment:
Should we limit the actions to only open search domain instead of all?

Reply:
Yes, we require access solely to the OpenSearch domain. The code has been updated accordingly.